$NetBSD: patch-aa,v 1.5 2021/09/28 09:20:34 ryoon Exp $

* Don't hardcode CC, CFLAGS, PREFIX. Fix installation stage.
* Add LDFLAGS to fix RELRO build.

--- Makefile.orig	2003-08-07 15:40:39.000000000 +0000
+++ Makefile
@@ -1,7 +1,6 @@
 ##################################################################
 # Directories
 # Install dirs
-INSTALL_PREFIX = /usr/local
 
 # Full default path is /usr/local/bin
 APG_BIN_DIR = /bin
@@ -60,12 +59,10 @@ APG_CLIBS += -lcrypt
 ##################################################################
 # You can modify CC variable if you have compiler other than GCC
 # But the code was designed and tested with GCC
-CC = gcc
 
 ##################################################################
 # Compilation flags
 # You should comment the line below for AIX+native cc
-CFLAGS = -Wall
 
 ####################################################################
 # If you plan to install APG daemon you should look at lines below #
@@ -129,13 +126,13 @@ cliserv: apgd apgbfm 
 cygwin: standalone
 
 apg:
-	${CC} ${CFLAGS} ${STANDALONE_OPTIONS} -o ${PROGNAME} ${SOURCES} ${APG_CLIBS}
+	${CC} ${CFLAGS} ${LDFLAGS} ${STANDALONE_OPTIONS} -o ${PROGNAME} ${SOURCES} ${APG_CLIBS}
 
 apgd:
-	${CC} ${CFLAGS} -DCLISERV ${CLISERV_OPTIONS} -o ${CS_PROGNAME} ${SOURCES} ${APG_CS_CLIBS}
+	${CC} ${CFLAGS} ${LDFLAGS} -DCLISERV ${CLISERV_OPTIONS} -o ${CS_PROGNAME} ${SOURCES} ${APG_CS_CLIBS}
 
 apgbfm:
-	${CC} ${FLAGS} -DAPGBFM -o ${BFM_PROGNAME} ${BFM_SOURCES} ${APGBFM_CLIBS}
+	${CC} ${CFLAGS} ${LDFLAGS} -DAPGBFM -o ${BFM_PROGNAME} ${BFM_SOURCES} ${APGBFM_CLIBS}
 
 strip:
 	strip ${PROGNAME}
@@ -143,24 +140,12 @@ strip:
 	strip ${BFM_PROGNAME}
 
 install:
-	if test -x ./apg; then \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_MAN_DIR}; \
-./install-sh -c -m 0755 -o root -g ${FIND_GROUP} ./apg ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./install-sh -c -m 0444 ./doc/man/apg.1 ${INSTALL_PREFIX}${APG_MAN_DIR}; \
-fi
-	if test -x ./apgd; then \
-./mkinstalldirs ${INSTALL_PREFIX}${APGD_BIN_DIR}; \
-./mkinstalldirs ${INSTALL_PREFIX}${APGD_MAN_DIR}; \
-./install-sh -c -m 0755 -o root -g ${FIND_GROUP} ./apgd ${INSTALL_PREFIX}${APGD_BIN_DIR}; \
-./install-sh -c -m 0444 ./doc/man/apgd.8 ${INSTALL_PREFIX}${APGD_MAN_DIR}; \
-fi
-	if test -x ./apgbfm; then \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_MAN_DIR}; \
-./install-sh -c -m 0755 -o root -g ${FIND_GROUP} ./apgbfm ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./install-sh -c -m 0444 ./doc/man/apgbfm.1 ${INSTALL_PREFIX}${APG_MAN_DIR}; \
-fi
+	${BSD_INSTALL_PROGRAM} apg ${DESTDIR}${PREFIX}/bin
+	${BSD_INSTALL_MAN} doc/man/apg.1 ${DESTDIR}${PREFIX}/${PKGMANDIR}/man1
+	${BSD_INSTALL_PROGRAM} apgd ${DESTDIR}${PREFIX}/sbin
+	${BSD_INSTALL_MAN} doc/man/apgd.8 ${DESTDIR}${PREFIX}/${PKGMANDIR}/man8
+	${BSD_INSTALL_PROGRAM} apgbfm ${DESTDIR}${PREFIX}/bin
+	${BSD_INSTALL_MAN} doc/man/apgbfm.1 ${DESTDIR}${PREFIX}/${PKGMANDIR}/man1
 
 install-cygwin:
 	if test -x ./apg.exe; then \
